nixos/modules/services/web-apps/coder.nix

```diff
@@ -0,0 +1,165 @@
+        description = lib.mdDoc ''
+          Coder uses a PostgreSQL database to store users, workspace metadata, and other deployment information
+        '';
+        default = null;
+        example = "postgresql://root:root@database/coder?sslmode=disable";
+      };
+
+      tlsCert = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          The path to the TLS certificate.
+        '';
+        default = null;
+      };
+
+      tlsKey = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          The path to the TLS key.
+        '';
+        default = null;
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.coder = let
+      # see https://github.com/coder/coder/issues/4731
+      pgCtl = pkgs.writeScript "pg_ctl" ''
+        #!${pkgs.bash}/bin/bash
+
+        exec ${pkgs.postgresql_14}/bin/pg_ctl "$@" -o '--unix_socket_directories=${cfg.homeDir}'
+      '';
+    in {
+      description = "Coder - Self-hosted developer workspaces on your infra";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        mkdir -p ${cfg.homeDir}/.config/coderv2/postgres/bin/bin
+        rm -f ${cfg.homeDir}/.config/coderv2/postgres/bin/bin/initdb
+        rm -f ${cfg.homeDir}/.config/coderv2/postgres/bin/bin/pg_ctl
+        rm -f ${cfg.homeDir}/.config/coderv2/postgres/bin/bin/postgres
+        rm -f ${cfg.homeDir}/.config/coderv2/postgres/bin/share
+        ln -s ${pkgs.postgresql_14}/bin/initdb ${cfg.homeDir}/.config/coderv2/postgres/bin/bin/initdb
+        ln -s ${pgCtl} ${cfg.homeDir}/.config/coderv2/postgres/bin/bin/pg_ctl
+        ln -s ${pkgs.postgresql_14}/bin/postgres ${cfg.homeDir}/.config/coderv2/postgres/bin/bin/postgres
+        ln -s ${pkgs.postgresql_14}/share ${cfg.homeDir}/.config/coderv2/postgres/bin/
+      '';
+
+      environment = {
+        CODER_ACCESS_URL = cfg.accessUrl;
+        CODER_WILDCARD_ACCESS_URL = cfg.wildcardAccessUrl;
+        CODER_PG_CONNECTION_URL = cfg.postgresqlUrl;
+        CODER_ADDRESS = cfg.listenAddress;
+        CODER_TLS_ENABLE = optionalString (cfg.tlsCert != null) "1";
+        CODER_TLS_CERT_FILE = cfg.tlsCert;
+        CODER_TLS_KEY_FILE = cfg.tlsKey;
+      };
+
+      serviceConfig = {
+        Type = "notify";
+        ProtectSystem = "full";
+        PrivateTmp = "yes";
+        PrivateDevices = "yes";
+        SecureBits = "keep-caps";
+        AmbientCapabilities="CAP_IPC_LOCK CAP_NET_BIND_SERVICE";
+        CacheDirectory="coder";
+        CapabilityBoundingSet="CAP_SYSLOG CAP_IPC_LOCK CAP_NET_BIND_SERVICE";
+        KillSignal="SIGINT";
+        KillMode="mixed";
+        NoNewPrivileges="yes";
+        Restart ="on-failure";
+        RestartSec=5;
+        TimeoutStopSec=90;
+        ExecStart = ''
+          ${cfg.package}/bin/coder server
+        '';
+        User = "coder";
+        Group = "coder";
```

Comment:
We probably should add an option so that it would be possible to set a different user/group.

Reply:
I will add it